packages/html-reporter/src/links.tsx

```diff
@@ -113,3 +113,36 @@ export function generateTraceUrl(traces: TestAttachment[]) {
 }
 
 const kMissingContentType = 'x-playwright/missing';
+
+type AnchorID = string | ((id: string | null) => boolean) | undefined;
+
+export function useAnchor(id: AnchorID, onChange: (isRevealed: boolean) => void) {
+  React.useEffect(() => {
+    const listener = () => {
```

Comment:
Is this going to trigger if I click the same "reveal" link again? I'd like the item to be revealed even when it's already in the url. And vice versa, I wouldn't want the item to be revealed on mere re-render, e.g. switching a tab in a tabbed pane.

Reply:
It triggers for each `popstate` event, unrelated of re-rendering. As you can see in the video, this includes clicking the link again. If you have a tabbed pane, and that tabbed pane leads to `popstate` events, then I guess the tabbed pane shouldn't include the `anchor` param in its links.